### PR TITLE
Customer access

### PR DESCRIPTION
**Pull request by NataMarkova**
@jrmcgarvey, please check it out. The PR consists of feature of access control for the customer (client) views and controller. The code that I worked on doesn't handle a case when a user has restriction to the CRUD operations in assignments based on his role.

### CODE REVIEW

File: app/controllers/customers_controller.rb [hunk 1]
```diff
@@ -1,12 +1,25 @@
...
+    # def index
+    #   @customers = Customer.all
+    # end
+
     def index
-      @customers = Customer.all
+      if current_user.role.nil?
+        flash.notice = "You are not authorized for that operation."
+        redirect_to home_index_path
+      elsif current_user.role.include?("volunteer")
```
Comment (jrmcgarvey) [inline app/controllers/customers_controller.rb:18]: This logic is not quite right.  It is possible for someone to be a volunteer and an administrator, in which case they should see the whole list.  It is also possible that the user has a role, but that role does not include volunteer or intern or administrator, in which case they are not authorized.

Something like:

if current_user.role.nil?
  not authorized
elsif current_user.role.include?('administrator') || current_user.role.include?('intern')
 show the full list
elsif current_user.role.include?('volunteer')
 show the current user's assigned list
else
 you are not authorized
end

File: app/controllers/customers_controller.rb [hunk 2]
```diff
@@ -1,12 +1,25 @@
...
+        @customers = current_user.customers
```
Comment (jrmcgarvey) [inline app/controllers/customers_controller.rb:19]: There should be access control added for each of the other operations, not just index.  Only administrators or interns could do new, edit, create, update, or delete.  Volunteers could do show, but only for customers to which they are assigned.  Users that don't have any of these roles can't do any of the operations.

**Review by jrmcgarvey — CHANGES_REQUESTED** [review]
Some changes needed -- let me know if you have questions.

File: app/controllers/customers_controller.rb (hunk 1 above)
Comment (NataMarkova) [reply app/controllers/customers_controller.rb:16]: Fixed

File: app/controllers/customers_controller.rb (hunk 2 above)
Comment (NataMarkova) [reply app/controllers/customers_controller.rb:17]: The logic was implemented.

**Comment by NataMarkova** [comment]
@jrmcgarvey , please check it out one more time

**Review by jrmcgarvey — APPROVED** [review]
OK, looks correct now!  I will merge.

**Pull request merged**